### ISSUE TITLE
Github open pull request count widget

### ISSUE THREAD

**Issue by timja** [spotify/backstage#1808]
## Feature Suggestion

Show the count of open pull requests for a repository (maybe even a breakdown with open/closed as well)

## Possible Implementation

A new annotation will likely be needed,
something like:

```
backstage.io/github-repo-slug: timja-org/my-repo
```

## Context

Surface more information in backstage, along with healthy repo indicators, i.e. less open PRs is better.

**Comment by Rugvip** [comment]
Sounds like a good idea!

Whoever picks this up, please make sure to sync with https://github.com/spotify/backstage/issues/1524 so that we don't end up with multiple annotations for the GitHub slug :grin:

**Comment by harshilparmar** [comment]
@Rugvip Can I try this?

**Comment by Rugvip** [comment]
@harshilparmar Sure, thanks!

**Comment by harshilparmar** [comment]
@Rugvip Thanks !! Can you elaborate more on this? I didn't get that clear idea about this.

**Comment by timja** [comment]
@harshilparmar 

My 2 cents on this:

I think it needs a new plugin, either 'github' or 'github-metrics' something like that.

It should use the GitHub API to show the number of open pull requests, possibly depending on the UI it could also show 'closed' and 'merged' PRs too.

To know the GitHub repo you will need a new annotation on the component and probably rename the annotation that the github actions widget has to use it.

Any more questions? or feel free to ping in discord as well.

**Comment by harshilparmar** [comment]
@timja Thanks you so much!! I will let you know 👍

**Comment by dtuite** [comment]
We are also working on a GitHub pull requests plugin [over here](https://github.com/RoadieHQ/backstage/pull/7). It's fairly basic so far but improving rapidly.

<img width="1025" alt="Screenshot 2020-08-08 at 12 03 25" src="https://user-images.githubusercontent.com/562403/89708466-60fcd100-d96f-11ea-8311-6896e0c98a2a.png">

We will include a pull requests count widget as a feature. If everything stays on track this should drop within a week or so. 

Let me know what you would like to see in the plugin please and we will get it done!

**Comment by timja** [comment]
PR open / closed / merge

Also not sure how do-able it would be but:

* average length of time a PR is open for
* average size of PR

Basically some of the stats from pullreminders

![image](https://user-images.githubusercontent.com/21194782/89708769-da95be80-d971-11ea-99d6-c5b1f146ee22.png)

**Comment by dtuite** [comment]
Things are progressing. Pagination, search etc. all works now. The next thing up will be filtering.

Once this PR is merged we will move the widgets to the tabs on the Entity overview page so you can look up pull requests for a specific service in the catalog.

<img width="1616" alt="github pull requests" src="https://user-images.githubusercontent.com/562403/89878917-1cb14100-dbba-11ea-8cf4-e21fbf7d638b.png">

@timja When you said "PR open / closed / merge" I presume you meant "Total number of PRs open / total number closed". What does "merge" mean?

**Comment by timja** [comment]
merged vs closed, but I'm not sure if it's too valuable

**Comment by harshilparmar** [comment]
@timja If  @dtuite working on this issue,I will take another issue.Is it ok?
